latex_access/polish_braille.py

```diff
@@ -0,0 +1,391 @@
+#
+#    This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation;
+#    either version 2 of the License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#    See the GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License along with this program; if not, visit <http://www.gnu.org/licenses>
+#
+'''Module to provide  translations to Polish  braille for the latex_access.'''
+
+from __future__ import absolute_import
+from latex_access import latex_access
+from latex_access.latex_access import get_arg
+import re
+
+
+sqrt_with_two_args =re.compile(r".*\\sqrt\[(.?)\]")
+
+class Braille(latex_access.translator):
+    '''Class for translations to braille.'''
+
+    def __init__(self):
+        latex_access.translator.__init__(self)
+        self.files.append("polish_braille.table")
+        self.load_files()
+        new_table={"\\dot":("","`"),"\\ddot":("","``"),
+                   "^":self.super,"_":self.sub,"\\sqrt":self.sqrt,"\\frac":self.frac,"\\dfrac":self.frac,"\\tfrac":self.frac,
+                   "\\tag":self.tag,"\\mathbf":("_",""),"\\mathbb":("_",""),"\\colvec":("{"," ","o"),"\\abs":(u"⠈l",u"⠸"),"\\tcolvec":("{"," "," ","o"),"\\bar":self.bar,"\\hat":self.bar,"\\widehat":self.bar,"\\overline":self.bar,",":self.comma,
+                   "L0":u"⠴","L1":u"⠂","L2":u"⠆","L3":u"⠒","L4":u"⠲","L5":u"⠢","L6":u"⠖",
+                   "L7":u"⠶","L8":u"⠦","L9":u"⠔","\\pmod":self.pmod,"\\log":self.log}
+
+        self.upperNumbers=('j','a','b','c','d','e','f','g','h','i')
+        for number in range (0,10): # add the numbers
+            new_table[str(number)]=self.numbers
+        for letter in range (65,91): # Ascii upper case
+            new_table ["%c" % (letter)] = self.upperLetter
+        for letter in range (97,123): # Ascii, lower case
+            new_table["%c" % (letter)] = self.lowerLetter
+
+        self.table.update(new_table)
+
+    sqrt_with_two_args =re.compile(r".*\\sqrt\[(.*)\]+.*")
+
+    def before (self):
+        """Method ran before the translator at depth = 0.
+
+        Place anything here to do before we begin translation to polish braille. Eg. set
+        variables etc. Should NOT alter the actual input at all."""
+        self.lastnumber = -1 # keep track of where the last latex number # was open i.e. after a digit . or , (for Polish braille)
+    
+    def super(self,input,start,rting=()):
+        '''Translate  superscripts into polish braille.
+
+        Returns a touple with translated string and index of
+        first char after end of super.'''
+        arg=get_arg(input,start)
+		#Handle squared, degrees  and cubed as special cases
```

Comment:
Since the rest of the file uses spaces for indentation this line should not be indented with tabs.